README.md

```diff
@@ -201,6 +202,26 @@ To enable teslausb to act as a wifi access point with the given SSID and passwor
    #export AP_IP='192.168.66.1'
    ```
 
+## Security
+
+As a little descussion of security for your Pi, please keep in mind the following items.
+
+1. If WiFi Access Point is configured, the AP password needs to be worth while. Make it something better than Passw0rd, more than 8 characters (odd number of upper/lower characters, numbers are the best.) 
```

Comment:
What exactly do you mean by "odd number" in this context? I don't think it really matters whether you use an odd or even number of any given type of character.

Reply:
Making a password an odd number of characters (actually a prime number of characters over 11) increases the difficulty of guessing a password by a good percentage. I always tell my users over 8 characters but an odd number (9, 11,13,...), upper/lower case letters and some numbers is best.

Reply:
Never heard that before, and it doesn't intuitively make sense to me (I would think "the more the better", so 11 is worse than 12, even though 11 is prime). Do you have a reference for that so I can educate myself?

Reply:
(e.g. while https://en.wikipedia.org/wiki/Password_strength talks about length, it only talks about greater length being better, and makes no mention of odd or prime lengths being better)

Reply:
I can't find the article on it but I remember helping the author do the research. (Its not under his name but it was done for the NSA so it may not be published.

The BEST way was to have a password manager generate a password of 13 characters long, Upper/Lower case, numbers and a selected set of special characters (that ALL systems accept) was the best that was usually needed. That would take a machine centuries to guess correctly if the user didn't change it at some time during that time. (Plus the hack attack would be caught by that time hopefully.)

We found that the non-prime number between 8-15 (8, 9, 10, 12, 14, 15) did not increase the time at the same rate as the prime numbers (11, 13).  However with there only being two of them, the study did not care that much.  For our study, we said that 12 to 13 characters, a number and adding a special character somewhere in that satisified the needs for 99% of the applications.

We then had to do the same just using words. We noted that unrelated words were best. With a dictionary of 10,000 words we got that 5 unrelated words gave us a 8 year crack perdiction. If you added numbers replacing some of the letters (NOT 0 = O or 3 = e)  or messing with the Upper/Lower case of the words would increase the guess time to almost 15 years.  Going to less than 5 words would reduce the total length of the password and bring the guess time into the 6 hours for 4 words and seconds for 3 words.

Reply:
Okay - Maybe I am doing this all wrong.....

What I did was create a fork of your repository.  I do comapirs metween my main-dev and your manin-dev to get updates others and you put in via a pull request. (That causes the update but makes it part of the next pull I issue to you. I don't understand that either.) I then use that fork to test updates by creating a branch and updating stuff from there using the github editor and commiting changes to the branch till it's cool. I then do a compare and create a pull request to me to merge it into my fork's main-dev.  When that looks cool, I do a compare of your main-dev and my main-dev creating a pull request.  Is that how I should be doing this?

Or should I be using git to download a local copy of the data, make changes there in a branch (somehow) and the create pull requests to my repository main-dev and the like. If you use git, it seems github does not see much of that - or I am doing things all wrong.  :-)

There does not seem to be a simple doc, that I have found, on how to be a contributor. Can you help out how I can contribute to you without messing with your repository.

Reply:
> The BEST way was to have a password manager generate a password of 13 characters long, Upper/Lower case, numbers and a selected set of special characters (that ALL systems accept) was the best that was usually needed. That would take a machine centuries to guess correctly if the user didn't change it at some time during that time. (Plus the hack attack would be caught by that time hopefully.)

But simple math dictates that using 14 characters instead of 13 would take N times longer for a brute force approach (where N is the total number of possible symbols, so with upper/lower/number/specials probably around 70 times or so). Each additional character increases the time it would take to crack by the same factor. There's nothing special about odd or prime numbers of characters in that regard.
 
In any case, in the absence of clear evidence I'd rather not mention anything about odd or prime password lengths being better than even or non-prime. Emphasizing the need for a strong password is good, but I don't really want to go into specifics other than perhaps "the longer the better". Feel free to link to https://en.wikipedia.org/wiki/Password_strength or https://xkcd.com/936/

Reply:
> Okay - Maybe I am doing this all wrong.....
> 
> What I did was create a fork of your repository. I do comapirs metween my main-dev and your manin-dev to get updates others and you put in via a pull request. (That causes the update but makes it part of the next pull I issue to you. I don't understand that either.) I then use that fork to test updates by creating a branch and updating stuff from there using the github editor and commiting changes to the branch till it's cool. I then do a compare and create a pull request to me to merge it into my fork's main-dev. When that looks cool, I do a compare of your main-dev and my main-dev creating a pull request. Is that how I should be doing this?

I'm not actually sure what the right/best way is. For documentation updates like this, it might be easiest to look at the README in my repo, then click the edit button at the top, and once you're done, select "Create a new branch for this commit and start a pull request" .

Reply:
I people use password vaults your assumption is correct. However, most users do not do that. The best they do is use a bunch of words connected together (i.e. Letme1n2thisSystem) but you can see that is much greater than 13 characters. That is why hackers look for words and construct phrases. That is why 5 unrelated words (i.e. StupidwormsareSpacecooks) are harder for systems to hack without setting off a password hack trap.

So how does this look for that section:

1. If WiFi Access Point is configured, the AP password needs to be worth while. Make it something better than Passw0rd, more than 8 characters. The longer the password the better. See [Here](https://en.wikipedia.org/wiki/Password_strength) for password strength.

Reply:
If that is cool, I will close and submit it again.

Reply:
Oops - should have been:

1. If WiFi Access Point is configured, the AP password needs to be worth while. Make it something better than Passw0rd, more than 8 characters. The longer the password the better. See [here](https://en.wikipedia.org/wiki/Password_strength) or [here](https://xkcd.com/936/) for password strength.

Reply:
Yeah, that looks good.

Reply:
Okay -  I will close this and put in another. I am going to try something a little different on this end.